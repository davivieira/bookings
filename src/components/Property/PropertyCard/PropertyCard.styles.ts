import { Card, CardActions, styled, Typography } from "@mui/material";

export const SCCardFooter = styled(CardActions)(() => ({
  display: 'flex',
  flexDirection: 'row',
  justifyContent: 'space-between',
  bottom: 0
}))

export const SCCard = styled(Card)(() => ({
  display: 'flex',
  flexDirection: 'column',
  justifyContent: 'space-between',
  height: '100%',
  width: '100%',
  maxWidth: 345
})) 

export const SCPrice = styled(Typography)(({ theme: { palette } }) => ({
  fontWeight: 'bold',
  color: palette.secondary.main
}));