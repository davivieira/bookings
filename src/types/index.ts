export type Property = {
  id: string
  userId: string
  name: string
  location: string
  description: string
  picture: string
  price: number,
  bookings: string[]
}

export type Booking = {
  id: string
  userId: string
  propertyId: string
  checkinDate: string
  checkoutDate: string
}

export type BookingResponse = {
  id: string
  userId: string
  property: Property
  checkinDate: string
  checkoutDate: string
}

export type User = {
  id: string
  name: string
  email: string
  properties: string[]
}

export type AppState = {
  user: {
    currentUser: User | null;
  };
}

