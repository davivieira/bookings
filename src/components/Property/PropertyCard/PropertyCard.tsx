import CardContent from '@mui/material/CardContent';
import CardMedia from '@mui/material/CardMedia';
import Typography from '@mui/material/Typography';
import { Property } from '@/types';
import { SCCard, SCCardFooter, SCPrice } from './PropertyCard.styles';
import { BookingModalGroup } from '@/components/Booking/';

type PropertyCardProps = {
  property: Property,
}

function PropertyCard({ property }: PropertyCardProps) {
  return (
    <SCCard>
      <CardMedia
        component="img"
        alt="green iguana"
        height="140"
        image={property.picture}
      />
      <CardContent>
        <Typography gutterBottom variant="h5" component="div">
          {property.name}
        </Typography>
        <Typography variant="body2" color="text.secondary">
          {property.description}
        </Typography>
      </CardContent>
      <SCCardFooter>
        <BookingModalGroup propertyId={property.id} />
        <SCPrice>U$ {property.price},00</SCPrice>
      </SCCardFooter>
    </SCCard>
  );
}

export default PropertyCard;