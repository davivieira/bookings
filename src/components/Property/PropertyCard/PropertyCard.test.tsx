import { render, screen } from "@testing-library/react";
import PropertyCard from "./PropertyCard";
import { Property } from "../../../types";

describe("PropertyCard", () => {
  const mockProperty: Property = {
    id: "0",
      userId: "0",
      name: "Luxury Lakeside Villa",
      location: "456 Lakefront Drive, Tahoe City, CA",
      price: 180,
      bookings: ["0"],
      description: "A luxurious lakeside villa offering breathtaking views of Lake Tahoe, private beach access, and elegant interiors.",
      picture: "https://images.unsplash.com/photo-1568605114967-8130f3a36994?fit=crop&w=800&q=80"
  };

  it("renders property card with correct information", () => {
    render(<PropertyCard property={mockProperty} />);

    const propertyNameElement = screen.getByText(/Luxury Lakeside Villa/i);
    expect(propertyNameElement).toBeInTheDocument();

    const propertyDescriptionElement = screen.getByText(
      /A luxurious lakeside villa/i
    );
    expect(propertyDescriptionElement).toBeInTheDocument();

    const propertyPriceElement = screen.getByText(/U\$ 180,00/i);
    expect(propertyPriceElement).toBeInTheDocument();
  });

  it("matches snapshot", () => {
    const { container } = render(<PropertyCard property={mockProperty} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
