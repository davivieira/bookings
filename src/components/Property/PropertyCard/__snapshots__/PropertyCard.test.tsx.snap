// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PropertyCard matches snapshot 1`] = `
<div
  class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-i1351c-MuiPaper-root-MuiCard-root"
>
  <img
    alt="green iguana"
    class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img css-o69gx8-MuiCardMedia-root"
    height="140"
    src="https://images.unsplash.com/photo-1568605114967-8130f3a36994?fit=crop&w=800&q=80"
  />
  <div
    class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
  >
    <div
      class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom css-h93ljk-MuiTypography-root"
    >
      Luxury Lakeside Villa
    </div>
    <p
      class="MuiTypography-root MuiTypography-body2 css-r40f8v-MuiTypography-root"
    >
      A luxurious lakeside villa offering breathtaking views of Lake Tahoe, private beach access, and elegant interiors.
    </p>
  </div>
  <div
    class="MuiCardActions-root MuiCardActions-spacing css-u3p8xk-MuiCardActions-root"
  >
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary css-1e6y48t-MuiButtonBase-root-MuiButton-root"
      tabindex="0"
      type="button"
    >
      Book
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
    <p
      class="MuiTypography-root MuiTypography-body1 css-1w4tqd2-MuiTypography-root"
    >
      U$ 
      180
      ,00
    </p>
  </div>
</div>
`;
